### PR TITLE
JAVA-1890: Add additional configuration for configuring SSLContext

### PR DESCRIPTION
**Pull request by tolbertam**
For [JAVA-1890](https://datastax-oss.atlassian.net/browse/JAVA-1890)

Motivation:

To enable providing keystore and truststore configuration for
DefaultSslEngineFactory via configuration instead of
relying on system properties to initialize the default
SSLContext.

Modifications:

Add configuration to DefaultDriverOption for configuring
SSLContext using keystore and truststore file location
and password.

Add tests for verifying configuration is effective.

Result:

DefaultSslEngineFactory can now build SSLContext from
configuration.

### CODE REVIEW

**Comment by tolbertam** [comment]
I'll also do [JAVA-1894](https://datastax-oss.atlassian.net/browse/JAVA-1894) on this PR if that's ok

File: manual/core/ssl/README.md [hunk 1]
```diff
@@ -101,22 +114,21 @@ passwords:
 
 If you need more control than what system properties allow, you need to write your own engine
 factory. If you just need specific configuration on the `SSLEngine`, you can extend the default
-factory and override `newSslEngine`. For example, here is how to enable hostname verification:
+factory and override `newSslEngine`. For example, here is how you would configure custom
+`AlgorithmConstraints`:
```
Comment (tolbertam) [inline manual/core/ssl/README.md:118]: This is a bit of a lame example now, couldn't think of anything else a user would want to configure.

Comment (olim7t) [reply manual/core/ssl/README.md:118]: That should be clear enough.

**Review by olim7t — APPROVED** [review]
LGTM. I think we can group all options in a single ticket/commit, I'll do the changes in jira.

**Comment by tolbertam** [comment]
> LGTM. I think we can group all options in a single ticket/commit, I'll do the changes in jira.

Sounds great, thanks!  I know there's a lot of merge activity going on right now.  When the time is right i can rebase and clean up the commit history into 1 commit, otherwise if you want to take that on feel free :+1:

**Comment by olim7t** [comment]
Done 😉 
I'm getting a sporadic test failure on `DefaultLoadBalancingPolicyIT.should_hit_non_replicas_when_routing_information_present_but_all_replicas_down`, but it's not related to this PR.

**Comment by tolbertam** [comment]
awesome, thanks!

**Pull request merged**